expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -299,13 +299,21 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                                 start = itr.getNextIdx();
                                 continue;
                             } else if (itr.peekNext() == ':') {
-                                if (flags.contains(Flag.DOUBLE_COLON) && itr.hasNext() && itr.peekNext() == ':') {
-                                    // TP 7
-                                    // OK actually the whole thing is really going to be part of the key
-                                    // Best approach is, rewind and do it over again, but without end-on-colon
-                                    itr.rewind(start);
-                                    keyNode = parseString(itr, !flags.contains(Flag.NO_RECURSE_KEY), true, false, flags);
-                                    list.add(new ExpressionNode(general, keyNode, Node.NULL));
+                                if (flags.contains(Flag.DOUBLE_COLON)) {
+                                    itr.next();
+                                    if (itr.hasNext() && itr.peekNext() == ':') {
+                                        // TP 7
+                                        // OK actually the whole thing is really going to be part of the key
+                                        // Best approach is, rewind and do it over again, but without end-on-colon
+                                        itr.rewind(start);
+                                        keyNode = parseString(itr, !flags.contains(Flag.NO_RECURSE_KEY), true, false, flags);
+                                        list.add(new ExpressionNode(general, keyNode, Node.NULL));
+                                    } else {
+                                        // TP 8
```

Comment:
This should be e.b. `TP 7.5` - or maybe `8.5` - the mnemonic means "test point" so that we can correlate tests with decision points in the NFA, even if they're similar. Then the corresponding test should be named accordingly so that it can be correlated.